guides/plugins/plugins/framework/data-handling/entities-via-attributes.md

```diff
@@ -0,0 +1,411 @@
+
+    #[Field(type: FieldType::TEXT)]
+    public ?string $text = null;
+
+    #[Field(type: FieldType::INT)]
+    public ?int $int;
+
+    // ...
+}
+```
+
+All field types are defined in the [`FieldType`](https://github.com/shopware/shopware/blob/trunk/src/Core/Framework/DataAbstractionLayer/Attribute/FieldType.php) class.
+
+We also provide a list of special field-types, which implement a specific behavior. They have an own php attribute class, for example, the `AutoIncrement` or `ForeignKey` field. 
+
+```php
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+    
+    #[AutoIncrement]
+    public int $autoIncrement;
+
+    #[ForeignKey(entity: 'currency')]
+    public ?string $foreignKey;
+}
+```
+
+## Json fields 
+
+If you want to store json data in a field, with an own validation and serialize logic, you can use the `Serialized` attribute, and define an own serializer class:
+
+```php
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+    
+    #[Serialized(serializer: PriceFieldSerializer::class)]
+    public ?PriceCollection $serialized = null;
+}
+```
+
+## Custom Fields
+
+If you want to allow custom fields, you can use the `CustomField` attribute. 
+
+```php
+<?php
+
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+    
+    /**
+     * @var array<string, mixed>
+     */
+    #[CustomFields]
+    public array $customFields;
+}
+
+```
+
+## Api encoding
+
+Each field of an entity is by default not exposed in the API. To expose a field in the API, you have to set the `api` parameter of the `Field` attribute to `true`, or to one of the scopes you want to allow. 
+
+```php
+<?php
+
+use Shopware\Core\Framework\Api\Context\AdminApiSource;
+use Shopware\Core\Framework\Api\Context\SalesChannelApiSource;
+
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID, api: true)]
+    public string $id;
+
+    #[Field(type: FieldType::STRING)]
+    public string $notExposed;
+
+    #[Field(type: FieldType::STRING, api: true)]
+    public string $everywhere;
+
+    #[Field(type: FieldType::STRING, api: [AdminApiSource::class])]
+    public string $adminOnly;
+
+    #[Field(type: FieldType::STRING, api: [SalesChannelApiSource::class])]
+    public string $storeOnly;
+```
+
+## Translated fields
+
+To also support shopware translations for your entity, you can set the `translated` property of the `Field` attribute to `true`. This will automatically create a `TranslatedField` for the field and
+registers a `EntityTranslationDefinition` for you.
+
+You can also define a `Translations` attribute on a property to allow loading all translations of the entity.
```

Comment:
```suggestion
To support Shopware translations for your entity, set the `translated` property of the `Field` attribute to `true`. This will automatically create a `TranslatedField` for the field and
register an `EntityTranslationDefinition` for you.

Additionally, you can define a `Translations` attribute on a property to enable loading all translations of the entity.
```